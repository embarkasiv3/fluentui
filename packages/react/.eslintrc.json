{
  // react-internal has some overrides we need to mirror
  "extends": ["../react-internal/.eslintrc"],
  "root": true
}
