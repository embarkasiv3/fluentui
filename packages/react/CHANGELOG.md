# Change Log - @fluentui/react

This log was last generated on Thu, 19 Mar 2020 17:52:34 GMT and should not be manually modified.
