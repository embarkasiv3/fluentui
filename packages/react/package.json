{
  "name": "@fluentui/react",
  "version": "8.0.0",
  "description": "Reusable React components for building web experiences.",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffects": [
    "lib/version.js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui"
  },
  "license": "MIT",
  "scripts": {
    "build": "just-scripts build",
    "bundle": "just-scripts bundle",
    "clean": "just-scripts clean",
    "code-style": "just-scripts code-style",
    "codepen": "node ../../scripts/local-codepen.js",
    "just": "just-scripts",
    "lint": "just-scripts lint",
    "start": "just-scripts dev:storybook",
    "start:legacy": "yarn workspace @fluentui/public-docsite-resources start",
    "start-test": "just-scripts jest-watch",
    "test": "just-scripts test",
    "update-snapshots": "just-scripts jest -u",
    "mf": "just-scripts mf"
  },
  "devDependencies": {
    "@fluentui/eslint-plugin": "^1.0.0",
    "@types/chalk": "^2.0.0",
    "@types/enzyme": "3.10.3",
    "@types/enzyme-adapter-react-16": "1.0.3",
    "@types/glob": "^7.1.1",
    "@types/node": "^10.3.2",
    "@types/react": "16.9.42",
    "@types/react-dom": "16.9.10",
    "@types/react-test-renderer": "^16.0.0",
    "@types/webpack-env": "1.16.0",
    "@fluentui/scripts": "^1.0.0",
    "@fluentui/example-data": "^8.0.0",
    "@fluentui/jest-serializer-merge-styles": "^8.0.0",
    "@fluentui/test-utilities": "^8.0.0",
    "@fluentui/webpack-utilities": "^8.0.0",
    "@fluentui/react-conformance": "^0.2.1",
    "chalk": "^2.1.0",
    "enzyme": "~3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "glob": "^7.1.2",
    "jest-snapshot": "~24.9.0",
    "office-ui-fabric-core": "^11.0.0",
    "react": "16.8.6",
    "react-app-polyfill": "~1.0.1",
    "react-dom": "16.8.6",
    "react-test-renderer": "^16.3.0"
  },
  "dependencies": {
    "@fluentui/date-time-utilities": "^8.0.0",
    "@fluentui/foundation-legacy": "^8.0.0",
    "@fluentui/font-icons-mdl2": "^8.0.0",
    "@fluentui/merge-styles": "^8.0.0",
    "@fluentui/react-focus": "^8.0.0",
    "@fluentui/react-hooks": "^8.0.0",
    "@fluentui/react-shared-contexts": "^1.0.0-beta.9",
    "@fluentui/react-window-provider": "^2.0.0",
    "@fluentui/set-version": "^8.0.0",
    "@fluentui/style-utilities": "^8.0.0",
    "@fluentui/theme": "^2.0.0",
    "@fluentui/utilities": "^8.0.0",
    "@microsoft/load-themed-styles": "^1.10.26",
    "tslib": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 <18.0.0",
    "@types/react-dom": ">=16.8.0 <18.0.0",
    "react": ">=16.8.0 <18.0.0",
    "react-dom": ">=16.8.0 <18.0.0"
  }
}
