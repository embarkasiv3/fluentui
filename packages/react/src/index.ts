import './version';
export * from '@fluentui/react-internal';
export * from './Breadcrumb';
export * from './Checkbox';
export * from './ComboBox';
export * from './DetailsList';
export * from './DocumentCard';
export * from './Dropdown';
export * from './GroupedList';
export * from './Link';
export * from './Pivot';
export * from './ShimmeredDetailsList';
export * from './Slider';
export * from './Toggle';
