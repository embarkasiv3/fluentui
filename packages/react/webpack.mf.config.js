const path = require('path');
const fs = require('fs');
const webpack = require('webpack');
const getResolveAlias = require('@fluentui/scripts/webpack/getResolveAlias');
const { createServeConfig } = require('@fluentui/scripts/webpack/webpack-resources');
const { webpackMerge } = require('just-scripts');

const shared = {
  react: { singleton: true },
  'react-dom': { singleton: true },
};

const rootComponents = fs
  .readdirSync(path.join(__dirname, 'src'))
  .filter(filename => filename[0].toUpperCase() === filename[0])
  .map(filename => filename.replace(/\.tsx?/, ''));
const rootComponentsExposes = {};
for (const component of rootComponents) {
  rootComponentsExposes[`./lib/${component}`] = `@fluentui/react/src/${component}`;
}

const myConfig = {
  output: {},
  entry: {},
  mode: 'development',
  devtool: 'cheap-module-source-map',
  optimization: {
    concatenateModules: false,
  },
  module: {
    rules: [],
  },
  plugins: [
    new webpack.container.ModuleFederationPlugin({
      name: 'fluentuiReact',
      filename: 'remoteEntry.js',
      exposes: {
        ...rootComponentsExposes,
        './lib/compat/Button': '@fluentui/react/src/compat/Button',
        '.': '@fluentui/react/src/index',
        './lib/index': '@fluentui/react/src/index',
      },
      shared,
    }),
  ],
  resolve: {
    alias: getResolveAlias(),
  },
  devServer: {
    port: 2345,
    contentBase: 'dist',
  },
};

const metaConfig = webpackMerge.merge(createServeConfig(), myConfig);

module.exports = metaConfig;
