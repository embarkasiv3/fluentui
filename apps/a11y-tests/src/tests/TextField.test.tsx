import * as React from 'react';
import { testComponent } from './testComponent';

import { TextFieldBasicExample } from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.Basic.Example';
import { TextFieldBorderlessExample } from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.Borderless.Example';
import { TextFieldControlledExample } from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.Controlled.Example';
import { TextFieldCustomRenderExample } from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.CustomRender.Example';
import { TextFieldErrorMessageExample } from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.ErrorMessage.Example';
import { TextFieldMultilineExample } from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.Multiline.Example';
// prettier-ignore
import { TextFieldPrefixAndSuffixExample }
  from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.PrefixAndSuffix.Example';
import { TextFieldStyledExample } from 'office-ui-fabric-react/lib/components/TextField/examples/TextField.Styled.Example';

export const textFieldTestComponents = [
  { name: 'TextField', pageName: 'TextFieldBasicExample', elem: <TextFieldBasicExample /> },
  { name: 'TextField', pageName: 'TextFieldBorderlessExample', elem: <TextFieldBorderlessExample /> },
  { name: 'TextField', pageName: 'TextFieldControlledExample', elem: <TextFieldControlledExample /> },
  { name: 'TextField', pageName: 'TextFieldCustomRenderExample', elem: <TextFieldCustomRenderExample /> },
  { name: 'TextField', pageName: 'TextFieldErrorMessageExample', elem: <TextFieldErrorMessageExample /> },
  { name: 'TextField', pageName: 'TextFieldMultilineExample', elem: <TextFieldMultilineExample /> },
  { name: 'TextField', pageName: 'TextFieldPrefixAndSuffixExample', elem: <TextFieldPrefixAndSuffixExample /> },
  { name: 'TextField', pageName: 'TextFieldStyledExample', elem: <TextFieldStyledExample /> }
];

textFieldTestComponents.forEach(component => testComponent(component));
