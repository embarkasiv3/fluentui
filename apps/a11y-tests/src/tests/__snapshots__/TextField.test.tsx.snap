// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runs Axe on TextField (TextFieldBasicExample) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<input type=\\"text\\" id=\\"TextField8\\" required=\\"\\" value=\\"\\" class=\\"ms-TextField-field field-6\\" aria-invalid=\\"false\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "#TextField8",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Form element does not have an implicit (wrapped) &lt;label>
- Form element does not have an explicit &lt;label>
- Element has no title attribute or the title attribute is empty",
      "text": "Fix any of the following: aria-label attribute does not exist or is empty. aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty. Form element does not have an implicit (wrapped) <label>. Form element does not have an explicit <label>. Element has no title attribute or the title attribute is empty.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "#TextField8",
      "ruleId": "label",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "label",
  },
]
`;

exports[`runs Axe on TextField (TextFieldBorderlessExample) 1`] = `Array []`;

exports[`runs Axe on TextField (TextFieldControlledExample) 1`] = `Array []`;

exports[`runs Axe on TextField (TextFieldCustomRenderExample) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<input type=\\"text\\" id=\\"TextField2\\" value=\\"\\" class=\\"ms-TextField-field field-6\\" aria-describedby=\\"TextFieldDescription3\\" aria-invalid=\\"false\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "#TextField2",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- Only title used to generate label for form element",
      "text": "Fix all of the following: Only title used to generate label for form element.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "#TextField2",
      "ruleId": "label-title-only",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "label-title-only",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<input type=\\"text\\" id=\\"TextField2\\" value=\\"\\" class=\\"ms-TextField-field field-6\\" aria-describedby=\\"TextFieldDescription3\\" aria-invalid=\\"false\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "#TextField2",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Form element does not have an implicit (wrapped) &lt;label>
- Form element does not have an explicit &lt;label>
- Element has no title attribute or the title attribute is empty",
      "text": "Fix any of the following: aria-label attribute does not exist or is empty. aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty. Form element does not have an implicit (wrapped) <label>. Form element does not have an explicit <label>. Element has no title attribute or the title attribute is empty.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "#TextField2",
      "ruleId": "label",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "label",
  },
]
`;

exports[`runs Axe on TextField (TextFieldErrorMessageExample) 1`] = `Array []`;

exports[`runs Axe on TextField (TextFieldMultilineExample) 1`] = `Array []`;

exports[`runs Axe on TextField (TextFieldPrefixAndSuffixExample) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">https://</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">https://</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".prefix-16 > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".prefix-16 > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">.com</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">https://</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">.com</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
]
`;

exports[`runs Axe on TextField (TextFieldStyledExample) 1`] = `Array []`;
